### ISSUE TITLE
Canned onboarding curl commands won't work when TLS is disabled

### ISSUE THREAD

**Issue by kderusso** [elastic/start-local#4]
The start-local script [disables TLS](https://github.com/elastic/start-local/blob/c2f2167516eeb293f94ed584a7c75dabbf94e374/start-local.sh#L285) for simpler use in development mode, so you need to use http instead of https.

Unfortunately, this has a side effect for users running the start-local script, in that onboarding to Kibana all of the copy-paste curl commands using `localhost:9200` use TLS and will return an error: 
![Image](https://github.com/user-attachments/assets/4ab78bcf-19ac-4c5e-9ec0-70c2cc8f9e98)

This seems like a poor onboarding experience, since all examples will need to be manually modified. Is there something we can do to make this more seamless? For example, could we detect whether SSL is enabled and update our curl commands, or should we revisit the decision to disable TLS?

**Comment by ezimuel** [comment]
I added the Elasticsearch endpoint in the output of start-local. Now it looks as follows:
```
🎉 Congrats, Elasticsearch and Kibana are successfully installed and running!"

🌐 Access Kibana at http://localhost:5601
Use 'elastic' as username and '...' as password.

🛠️ Configuration details
We created the folder 'elastic-start-local' containing the following files:
  - docker-compose.yml: Use this file to manage the services.
  - .env: This file contains environment variables and credentials.
Learn more at https://github.com/elastic/start-local

🔑 An API key for Elasticsearch has been created (stored in .env):
Tm9a3e4d...==

ℹ️ Use this API key to connect to Elasticsearch (http://localhost:9200)
Using cURL you can test the connection with the command:
curl http://localhost:9200 -H 'Authorization: ApiKey Tm9a3e4d...=='
Learn more about our SDK at https://www.elastic.co/guide/en/elasticsearch/client
```

**Comment by ezimuel** [comment]
We fixed the Elasticsearch url in Kibana, thanks to this PR https://github.com/elastic/kibana/pull/193716. We just added the `ELASTICSEARCH_PUBLICBASEURL` variable to set the new endpoint to `http://localhost:9200`. Kibana displays the new URL in the Search page and in the client examples. 
This new ENV variable will be available in 8.16.0. @kderusso this should fix the issue.

**Comment by kderusso** [comment]
@ezimuel That's great!! Thank you for taking care of this, I think it'll be a huge UX improvement! 👏
